userProfile/src/components/app/app.tsx

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable prettier/prettier */
+import React, { useState } from 'react';
+import CustomModal from '../custom-modal/custom-modal';
+import Form from '../form/form';
+import { FormState } from '../../types';
+
+const CITIES = [`Москва`, `Санкт-Петербург`, `Екатеринбург`, `Новосибирск`, `Владивосток`, `Иннополис`];
+
+const initialState: FormState = {
+    firstname: ``,
+    lastname: ``,
+    city: ``
+};
+
+const App = () => {
+    const [state, setState] = useState(initialState);
+    const [prevState, setPrevState] = useState(initialState);
+    const [isModalOpen, setIsModalOpen] = useState(false);
+    const [modalMsg, setModalMsg] = useState<string[]>([]);
+
+    const getModalMessage = (state: FormState, prevState: FormState) => {
+        const result = [];
+        for (const key of Object.keys(state) as (keyof FormState)[]) {
+            if (state[key] !== prevState[key]) {
+                let message = ``;
+                switch (key) {
+                    case `firstname`:
+                        message += `Имя: `;
+                        break;
+                    case `lastname`:
+                        message += `Фамилия: `;
+                        break;
+                    case `city`:
+                        message += `Город: `;
+                        break;
+                }

```

Comment:
Можно очень просто избежать этого `switch`, добавив вначале файла такую конструкцию

<img width="362" alt="Снимок экрана 2021-03-09 в 11 40 40" src="https://user-images.githubusercontent.com/31249664/110429161-51dc2e00-80cc-11eb-9606-73d5515b38bd.png">

И тогда в этом месте получится 
```suggestion
let message = `${FormDataNames[key]}: `;
```
Минус `switch` плюс удобство получения имен полей в дальнейшем. Кайф!

Reply:
Думал в эту сторону, но не успел докрутить.